qi-lib/flow/core/deforest.rkt

```diff
@@ -129,6 +129,15 @@
          [else
           (λ (ctx name) #'(λ (v) v))])]))
 
+  (define-syntax-class cad*r-datum
+    #:attributes (countdown)
+    (pattern (~datum car) #:attr countdown #'0)
+    (pattern (~datum cadr) #:attr countdown #'1)
+    (pattern (~datum caddr) #:attr countdown #'2)
+    (pattern (~datum cadddr) #:attr countdown #'3)
+    (pattern (~datum caddddr) #:attr countdown #'4)
+    (pattern (~datum cadddddr) #:attr countdown #'5))
```

Comment:
This all makes sense, but seems unsustainable for other c[ad]+r forms.

Reply:
Other forms like `cadar` are not deforestable consumers. These are all deforestable because they terminate the fused sequence.